### PR TITLE
Add implement outcome scoring to compare

### LINKED ISSUE #408: [P2] No implement-task outcome scoring or populated implement cells (follow-up to #332)
## Priority
P2

## Type
Feature / Evidence infrastructure · **Target: 0.28.0 (minor — new measured capability)**

## Area
`src/infrastructure/compare.ts` (`benchmark_outcome`/`claim_assessment` are explain-shaped), `src/infrastructure/benchmark/suite.ts` (generic `task` plumbing + `--trials` already shipped), `src/runtime/implementation-pack.ts`.

## Problem
The CLI advertises `--task explain|implement|review|impact`, and #332's suite scaffolding shipped with a `task` field and `--trials`. But every measured outcome is explain-shaped: `benchmark_outcome` / `claim_assessment` score routing/tokens/turns, which fit answer-from-pack explain work, not edit work. There is no `implement_outcome` analogue and no populated implement cells. Madar's measured surface is half its advertised surface.

Project risk note (CLAUDE.md): *"Implementation tasks need edit files, test files, contracts, validation commands, and risk boundaries, not only explanation context."* #358 explicitly lists implement as "not yet measured."

## Desired behavior
- Add an `implement_outcome` block to compare reports for `--task implement`, scoring edit-relevant dimensions:
  - `edit_targeting`: did the pack's recommended edit files overlap the diff the agent produced?
  - `test_coverage`: did the pack surface the test file(s) the agent needed?
  - `risk_boundary_respect`: did the agent touch files the pack flagged under risk boundaries?
- Populate ≥1 implement cell in the suite (public-proxy repo, per #332's no-private-repo rule).
- Score on touched-file overlap + tests-still-pass, not diff equality.

## Acceptance criteria
- [ ] `madar compare ... --task implement` emits an `implement_outcome` block with the three dimensions.
- [ ] At least one implement cell is populated under the suite results with its artifacts.
- [ ] Reports without implement measurement explicitly show `implement_outcome: not_measured` so users aren't misled.
- [ ] Fixture test for `implement_outcome` classification.

## Validation
```bash
cd /Users/mohammednaji/Desktop/projects/works/madar
npm run typecheck && npm run build && CI=1 npm run test:run
madar pack "Add field X to endpoint Y" --task implement --why
madar compare "Add field X to endpoint Y" --task implement --baseline-mode native_agent --yes
```

## Non-goals
- Do not score on exact diff equality.
- Do not require an LLM judge (deterministic dimensions first).
- review/impact outcome scoring is related future work, tracked separately.

## Related
- #332 (closed) — benchmark suite scaffolding (task set includes implement; only explain populated).
- #358 (closed) — explicitly lists implement as "not yet measured."

## Suggested labels
priority:p2, type:feature, type:benchmark

### PR DESCRIPTION
**Pull request by mohanagy**
## Summary
- add compare task plumbing and benchmark-suite task propagation for implement-mode runs
- score native-agent implement compares with isolated per-arm workspaces, touched-file evidence, validation results, and implement_outcome reporting
- add issue-408 regressions for compare CLI, benchmark suite, and implement outcome classification

## Testing
- npm run typecheck
- npm run build
- CI=1 npm run test:run

Closes #408

## Summary by CodeRabbit

* **New Features**
  * Added "implement" task type for benchmark suite
  * Added `--task` option to compare command for selecting task type (defaults to "explain")
  * Implement task includes validation checks for code modifications, test coverage, and risk assessment

[Review Change Stack](https://app.coderabbit.ai/change-stack/mohanagy/madar/pull/413?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tests/unit/benchmark-suite.test.ts [hunk 1]
```diff
@@ -507,6 +511,115 @@ describe('runBenchmarkSuite', () => {
...
+                        baselineGrep: 1,
+                        madarGrep: 1,
+                      })
+                    },
+                  },
+                )
+
+                expect(seenTasks).toEqual(['implement', 'implement'])
+              })
+            })
```
[the remark this answers is not in the document]
Comment (mohanagy) [reply tests/unit/benchmark-suite.test.ts:622]: Verified against the current file: this test is already top-level, and `expect(seenTasks).toEqual(["implement", "implement"])` runs after the awaited `runBenchmarkSuite(...)` call. There is no unreachable `it(...)` inside `executeNativeAgentCompare` in the current revision, so I am resolving this thread as stale.
